### PR TITLE
Initial Implementation of environment field in connections

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

This PR adds a environment modal for future implementation for adding connections to an environment where any particular environment is not found then we can create also

**Currently I have made a sample modal for creating environment, when we will have actual fields we can replace it and improve styling accordingly, Right now it will be commented out** 

https://github.com/meshery/meshery/assets/90546692/c89ce71b-f624-482c-b118-7a601acdcc6f

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/connections/environment.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  autocompleteOptions : {
+    "& .MuiAutocomplete-option" : {
+      background : theme.palette.secondary.mainBackground,
+    },
+  },
+  dialogTitle : {
+    textAlign : 'center',
+    minWidth : 400,
+    padding : '10px',
+    color : "#fff",
```
Comment (sudhanshutech) [inline ui/components/connections/environment.js:32]: I will add a theme reference here too

**Comment by leecalcote** [comment]
After the user has chosen an environment and shifted focus off of the dropdown field, the "Choose an Environment" label should go away.

**Comment by Ghat0tkach** [comment]
hey @sudhanshutech 
 Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by leecalcote** [comment]
@senali-d look good?

**Comment by leecalcote** [comment]
Merge conflict.

**Comment by leecalcote** [comment]
@senali-d, will you take this PR forward?

**Comment by sudhanshutech** [comment]
@senali-d i have removed modal and create new env without modal. You can take a look and suggest changes if needed.

**Comment by leecalcote** [comment]
@sudhanshutech is autocomplete being used for creation of a new environment? Are you allowing connections to belong to one of more environments?

**Comment by sudhanshutech** [comment]
> @sudhanshutech is autocomplete being used for creation of a new environment? Are you allowing connections to belong to one of more environments?

Yes it can will help to create new env if it will not in options and environments are multiselect so more than one env can belong to a connection

**Pull request closed without merging**